File: src/ui/src/models/api/stream/Request/RequestEnd.ts
import { BaseStreamItem } from '../BaseStreamItem';

/**
 * Message to indicate the end of a request
 */
export interface RequestEnd extends BaseStreamItem {
  /**
   * The type of message - always 'RequestEnd'
   */
  MessageType: 'RequestEnd';
  
  /**
   * The conversation ID created or used for this chat session
   */
  ConversationId?: string;
}

File: src/lib/DonkeyWork.Chat.Common/Models/Streaming/Request/RequestEnd.cs
// ------------------------------------------------------
// <copyright file="RequestEnd.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Common.Models.Streaming.Request;

/// <summary>
/// A request end.
/// </summary>
public record RequestEnd : BaseStreamItem
{
    /// <summary>
    /// Gets the duration of the request.
    /// </summary>
    public TimeSpan Duration { get; init; }

    /// <summary>
    /// Gets the conversation id.
    /// </summary>
    public Guid? ConversationId { get; init; }
}